Odata-docs/webapi-8/fundamentals/client-driven-paging.md

```diff
@@ -0,0 +1,187 @@
+---
+title:  "Client-driven Paging in ASP.NET Core OData 8"
+description: Client-driven paging in ASP.NET Core OData 8.
+date:   2023-01-10
+ms.date: 1/10/2023
+author: habbes
+ms.author: clhabins
+---
+
+# Client-driven Paging in ASP.NET Core OData 8
+
+**Applies To**:[!INCLUDE[appliesto-webapi](../../includes/appliesto-webapi-v8.md)]
+
+The client can request the OData service to return a specific number of results. The `$top` query option is used to limit the number of results and the `skip` query option used to specify the offset. For example `GET /Customers?$skip=3$top=5` will return up to 5 items starting from the 4th item in the collection (i.e. after skipping the first 3).
+
+Using these two query options the client can divide a large collection into separate pages and fetch the pages in separate requests. For example, if our collection has 25 items, and we want each page to have 10 items, then we can fetch the different page as follows:
+
+- First page: `$top=10`
+- Second page: `$skip=10&$top=10`
+- Third page: `$skip=20&top=10` (The last page will return only 5 items)
+
+By default, ASP.NET Core OData 8 limits the maximum value for `$top` to 0. So if you add `$top=2` to your request, you would get an error with a message like:
+
+```json
+{
+    "error": {
+        "code": "",
+        "message": "The query specified in the URI is not valid. The limit of '0' for Top query has been exceeded. The value from the incoming request is '2'."
+}
+```
+
+You can configure the maximum top value using the [`ODataOptions.SetMaxTop()`](/dotnet/api/microsoft.aspnetcore.odata.odataoptions.setmaxtop) when adding OData services to your application. The following snippet would set the maximum allowed `$top` value to 100.
+
+```csharp
+services.AddOData(options =>
+    options.SetMaxTop(100)
+    //...
+    );
+```
+
+Now this will support values of up to 100, but return an error if `$top` exceeds 100.
+
+You can also remove the limit by setting it to `null`:
+
+```csharp
+services.AddOData(options =>
+    options.SetMaxTop(null)
+    // ...
+    );
+```
+
+Next, we add the [`EnableQuery`](/dotnet/api/microsoft.aspnetcore.odata.query.enablequeryattribute) attribute to the controller action that handles the request. This attribute automatically applies the OData query options in the request to the results returned by the controller action.
+
+```csharp
+public class ProductsController : ODataController
+{
+    [EnableQuery]
+    public IQueryable<Product> Get()
+    {
+        return productsCollection;
+    }
+}
+```
```

Comment:
@habbes I don't feel very strongly about this but based on how you have presented this section, I feel that it's a bit hard for someone to know that configuring client-side paging involves 2 steps.
I feel like a reorg of the section could help, starting with a heading level 2 "Configuring client-side paging | Configuring client-side paging in OData".
Then, the next line could be "Configuring client-side paging involves two steps:". Then a numbered list comprising of first, configuring the relevant OData services/features and second about the `EnableQuery` attribute. That way, it'd be clear that the two steps are required.
Some people have a tendency of trying things out as they read and without the indication that it's a two-step process, they could get stuck wondering why it doesn't work after performing step 1 only.